### DISCUSSION TITLE
Split tunneling on Android

### DISCUSSION THREAD

**Discussion by karnilamit** [vpnhood/VpnHood#26] [Q&A] [answered]
Hi,
I like the concept of this product and should appreciate you for making it available. I was curious to understand how do we achieve split tunneling in Vpnhood. I need a mechanism in Android that allows certain traffic to go through Vpnhood and other traffic being routed the normal route. I believe this is possible, correct?

Thanks

**Comment by trudyhood** [comment, marked as answer]
Yes, it is possible. We have already implemented it in VpnHood. In android, you can split by App and By IpRange. There is no limitation on android. Split by app is not supported by windows.
VpnHood always split Local Network, for custom IpRange you need to give your ipRange manually. You can also split by country which is actually a group of IpRange.

Let me know if you don't get your answer.

**Reply by karnilamit** [reply]
Thanks this helps. On Windows, I believe, since you are using Windivert it should also be theoretically possible since Windivert provides app info for each connection is that correct?

**Reply by trudyhood** [reply]
I am not sure about that. at least windivert doesn't support App info on WINDIVERT_LAYER_NETWORK. Maybe it is possible with WINDIVERT_LAYER_SOCKET. We don't have plan to work on App Splitting with WinDivert.
I am thinking to use Windows.Networking.Vpn on windows 10, but unfortunately couldn't find any working sample for it yet.

**Reply by tkmw1985** [reply]
Actually, split tunneling in VpnHood implemented not correctly it shows  gallary i.t.c  unessesery junk of restrictive apps (small number of apps not full android apps sorry for my bad english !!!! )))

**Reply by trudyhood** [reply]
@tkmw1985 Thank you for checking it out. Would you let me know more details of the problem. Can you tell me what type of application does not show in the list or let me know an example?

**Reply by tkmw1985** [reply]
Hello.  as you can see in the screenshot there  are useless applications, other applications are not on the list for example youtube, facebook so on !


![Screenshot_20210807-225100_VpnHood!](https://user-images.githubusercontent.com/30529245/128609678-c710ea81-5e0c-4aa7-b966-86b1f8b3c167.jpg)
![Screenshot_20210807-225047_VpnHood!](https://user-images.githubusercontent.com/30529245/128609680-9b90d447-5ed1-4a41-9a39-ac6bf97de11b.jpg)
![Screenshot_20210807-225031_VpnHood!](https://user-images.githubusercontent.com/30529245/128609682-c0000d90-1e39-404e-a06b-5d01a2cdf92a.jpg)

**Reply by trudyhood** [reply]
Thank you for reporting this. It looks we should use another android API the get the list of Apps. I hope it gets fix in another release. 
I opened an issue for this, and will inform you as soon as we updated it.
https://github.com/vpnhood/VpnHood/issues/31

Feel free to report us any more issue.

**Reply by trudyhood** [reply]
Dear @tkmw1985
The issue has been resolved in version v2.0.271
